openapi3filter/param_decoder_test.go

```diff
@@ -0,0 +1,1041 @@
+						Parameter: &openapi3.Parameter{Name: "X-Param", In: "cookie", Style: "form", Explode: noExplode, Schema: arrayOf(integerSchema)},
+						Reason:    "an invalid value",
+					},
+				},
+				{
+					name:   "invalid number items",
+					param:  &openapi3.Parameter{Name: "X-Param", In: "cookie", Style: "form", Explode: noExplode, Schema: arrayOf(numberSchema)},
+					cookie: "X-Param:1.1,foo",
+					err: &RequestError{
+						Parameter: &openapi3.Parameter{Name: "X-Param", In: "cookie", Style: "form", Explode: noExplode, Schema: arrayOf(numberSchema)},
+						Reason:    "an invalid value",
+					},
+				},
+				{
+					name:   "invalid boolean items",
+					param:  &openapi3.Parameter{Name: "X-Param", In: "cookie", Style: "form", Explode: noExplode, Schema: arrayOf(booleanSchema)},
+					cookie: "X-Param:true,foo",
+					err: &RequestError{
+						Parameter: &openapi3.Parameter{Name: "X-Param", In: "cookie", Style: "form", Explode: noExplode, Schema: arrayOf(booleanSchema)},
+						Reason:    "an invalid value",
+					},
+				},
+			},
+		},
+		{
+			name: "cookie object",
+			testCases: []testCase{
+				{
+					name:   "form",
+					param:  &openapi3.Parameter{Name: "X-Param", In: "cookie", Style: "form", Explode: noExplode, Schema: objectSchema},
+					cookie: "X-Param:id,foo,name,bar",
+					want:   map[string]interface{}{"id": "foo", "name": "bar"},
+				},
+				{
+					name:   "invalid integer prop",
+					param:  &openapi3.Parameter{Name: "X-Param", In: "cookie", Style: "form", Explode: noExplode, Schema: objectOf("foo", integerSchema)},
+					cookie: "X-Param:foo,bar",
+					err: &RequestError{
+						Parameter: &openapi3.Parameter{Name: "X-Param", In: "cookie", Style: "form", Explode: noExplode, Schema: objectOf("foo", integerSchema)},
+						Reason:    "an invalid value",
+					},
+				},
+				{
+					name:   "invalid number prop",
+					param:  &openapi3.Parameter{Name: "X-Param", In: "cookie", Style: "form", Explode: noExplode, Schema: objectOf("foo", numberSchema)},
+					cookie: "X-Param:foo,bar",
+					err: &RequestError{
+						Parameter: &openapi3.Parameter{Name: "X-Param", In: "cookie", Style: "form", Explode: noExplode, Schema: objectOf("foo", numberSchema)},
+						Reason:    "an invalid value",
+					},
+				},
+				{
+					name:   "invalid boolean prop",
+					param:  &openapi3.Parameter{Name: "X-Param", In: "cookie", Style: "form", Explode: noExplode, Schema: objectOf("foo", booleanSchema)},
+					cookie: "X-Param:foo,bar",
+					err: &RequestError{
+						Parameter: &openapi3.Parameter{Name: "X-Param", In: "cookie", Style: "form", Explode: noExplode, Schema: objectOf("foo", booleanSchema)},
+						Reason:    "an invalid value",
+					},
+				},
+			},
+		},
+	}
+
+	for _, tg := range testGroups {
+		t.Run(tg.name, func(t *testing.T) {
+			for _, tc := range tg.testCases {
+				t.Run(tc.name, func(t *testing.T) {
+					req, err := http.NewRequest(http.MethodGet, "http://test.org/test"+tc.path, nil)
+					if err != nil {
```

Comment:
From here and until the end of the file, could you use `"github.com/stretchr/testify/require"` instead?

Reply:
See here https://github.com/getkin/kin-openapi/blob/e6aa6352832d5c2b0bc28bc5e1065db6cc18d221/openapi3/schema_test.go#L52 how it makes the assumptions a bit more readable.

Reply:
Done.